replica/tablets.cc

```diff
@@ -461,10 +477,8 @@ future<std::optional<tablet_transition_stage>> read_tablet_transition_stage(cql3
 }
 
 stop_iteration tablet_sstable_set::for_each_sstable_set_until(const dht::partition_range& pr, std::function<stop_iteration(lw_shared_ptr<sstables::sstable_set>)> func) const {
-    size_t candidate_start = pr.start() ? group_of(pr.start()->value().token()) : size_t(0);
-    size_t candidate_end = pr.end() ? group_of(pr.end()->value().token()) : (_sstable_sets.size() - 1);
-    for (auto i = candidate_start; i <= candidate_end; i++) {
-        if (const auto& set = _sstable_sets[i]) {
+    for (const auto& i : subrange(pr)) {
+        if (const auto& set = find_sstable_set(i)) {
```

Comment:
if `find_sstable_set(i)` always returns an engaged lw_shared_ptr we don't need to the condition, do we?

Reply:
yes, I am planning to remove it later to reduce changes in the patch.